sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ChangeFeedQueryImpl.java

```diff
@@ -158,6 +158,19 @@ private Mono<FeedResponse<T>> executeRequestAsync(RxDocumentServiceRequest reque
 
                              final FeedResponse<T> feedResponse = BridgeInternal.toChangeFeedResponsePage(
                                  rsp, this.factoryMethod, klass);
+
+                             Map<String, String> rspHeaders = feedResponse.getResponseHeaders();
+                             String requestPkRangeId = null;
+                             if (!rspHeaders.containsKey(HttpConstants.HttpHeaders.PARTITION_KEY_RANGE_ID) &&
+                                 (requestPkRangeId = request
+                                     .getHeaders()
+                                     .get(HttpConstants.HttpHeaders.PARTITION_KEY_RANGE_ID)) != null) {
+
+                                 rsp.getResponseHeaders().put(
```

Comment:
I think it works fine, but should we use feedResponse.getResponseHeaders (just in case we decided to clone the map instead someday directly instead of reference the header)

Reply:
if getResponseHeaders would clone even that wouldn't help because we would add the header only to the cloned map returned from that call. We would always need an overload which doesn't clone. Bt line 169 should use the previosuly retrieved map instead of calling getResponseHeaders again - so changed that.